File: src/MiniAdmin.Domain/Entities/AlertRule.cs
namespace MiniAdmin.Domain.Entities;

public sealed class AlertRule
{
    public Guid Id { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Metric { get; set; } = string.Empty;

    public string Operator { get; set; } = ">=";

    public decimal Threshold { get; set; }

    public int WindowMinutes { get; set; } = 1440;

    public string Level { get; set; } = "Warning";

    public bool Enabled { get; set; } = true;

    public bool NotifyEnabled { get; set; } = true;

    public bool EmailEnabled { get; set; }

    public List<AlertRuleRecipient> Recipients { get; set; } = [];

    public int Sort { get; set; }

    public string? Remark { get; set; }

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: frontend/vue-vben-admin/apps/web-antd/src/api/system/alert-rule.ts
import { requestClient } from '#/api/request';

export interface AlertRuleItem {
  code: string;
  createdAt: string;
  description: string;
  emailEnabled: boolean;
  enabled: boolean;
  id: string;
  level: string;
  metric: string;
  name: string;
  notifyEnabled: boolean;
  operator: string;
  recipients: AlertRuleRecipient[];
  remark?: null | string;
  sort: number;
  threshold: number;
  updatedAt: string;
  windowMinutes: number;
}

export interface AlertRuleRecipient {
  id: string;
  recipientId: string;
  recipientName: string;
  recipientType: 'Role' | 'User';
}

export interface AlertRuleListParams {
  enabled?: boolean;
  keyword?: string;
  level?: string;
  page?: number;
  pageSize?: number;
}

export interface AlertRuleListResult {
  items: AlertRuleItem[];
  total: number;
}

export interface UpdateAlertRuleParams {
  emailEnabled: boolean;
  enabled: boolean;
  level: string;
  notifyEnabled: boolean;
  recipientRoleIds: string[];
  recipientUserIds: string[];
  remark?: null | string;
  threshold: number;
  windowMinutes: number;
}

export async function getAlertRuleListApi(params: AlertRuleListParams) {
  return requestClient.get<AlertRuleListResult>('/system/alert-rule/list', {
    params,
  });
}

export async function updateAlertRuleApi(
  id: string,
  data: UpdateAlertRuleParams,
) {
  return requestClient.put<AlertRuleItem>(`/system/alert-rule/${id}`, data);
}
